lib/core/base/context.js

```diff
@@ -45,7 +46,9 @@ function pushUniqueFrameSelector(context, type, selectorArray) {
   context.frames = context.frames || [];
 
   const frameSelector = selectorArray.shift();
-  const frames = document.querySelectorAll(frameSelector);
+  const frames = querySelectorAll(context.flatTree, frameSelector).map(
```

Comment:
This introduced quite a few problems:

1. Existing uses of include/exclude are going to find more nodes than it does today, because they are now applied over all trees.
2. Real CSS selectors will find the wrong element. `some-component > :first-child` will get the first child in the flat tree, instead of the first slotted element.
3. Selectors currently supported won't be. utils.querySelectorAll only supports a subset of CSS
4. IDs selection becomes very quirky, since they're no longer used

This has the potential to break a fair amount of existing scripts running axe-core today. I'm not in favor of this change.